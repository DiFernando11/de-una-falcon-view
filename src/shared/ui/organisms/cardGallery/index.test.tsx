import { render, screen } from '@testing-library/react';
import CardGallery from './index';
import type { CardProps } from '../../molecules/card/types';

const exampleCards: CardProps[] = [
  {
    id: 'card-1',
    theme: 'glassDark',
    children: <span>Card 1</span>,
  },
  {
    id: 'card-2',
    theme: 'glassWhite',
    children: <span>Card 2</span>,
  },
  {
    id: 'card-3',
    theme: 'glassDark',
    children: <span>Card 3</span>,
  },
];

describe('<CardGallery />', () => {
  it('should render the correct number of cards', () => {
    // Arrange & Act
    render(<CardGallery cards={exampleCards} minCardWidth={200} />);
    // Assert
    expect(screen.getByText('Card 1')).toBeInTheDocument();
    expect(screen.getByText('Card 2')).toBeInTheDocument();
    expect(screen.getByText('Card 3')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    // Arrange & Act
    const { asFragment } = render(<CardGallery cards={exampleCards} minCardWidth={180} />);
    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});
