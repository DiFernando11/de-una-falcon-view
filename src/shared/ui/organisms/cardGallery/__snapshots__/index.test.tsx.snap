// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<CardGallery /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex gap-2 flex-wrap justify-center items-stretch w-full"
  >
    <div
      class="flex-1 w-full flex flex-col"
      style="min-width: 180px; max-width: 300px;"
    >
      <div
        class="rounded-2xl p-6 transition-all relative overflow-hidden shadow-lg bg-gradient-to-br border cursor-pointer card-hover glassBlurDark flex-1"
        id="card-1"
      >
        <span>
          Card 1
        </span>
      </div>
    </div>
    <div
      class="flex-1 w-full flex flex-col"
      style="min-width: 180px; max-width: 300px;"
    >
      <div
        class="rounded-2xl p-6 transition-all relative overflow-hidden shadow-lg bg-gradient-to-br border cursor-pointer card-hover glassBlurWhite flex-1"
        id="card-2"
      >
        <span>
          Card 2
        </span>
      </div>
    </div>
    <div
      class="flex-1 w-full flex flex-col"
      style="min-width: 180px; max-width: 300px;"
    >
      <div
        class="rounded-2xl p-6 transition-all relative overflow-hidden shadow-lg bg-gradient-to-br border cursor-pointer card-hover glassBlurDark flex-1"
        id="card-3"
      >
        <span>
          Card 3
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
